test/fabrik.VirtualHostAgent.spec.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const lib = require('../lib');
+const VirtualHostAgent = lib.fabrik.VirtualHostAgent;
+
+describe('fabrik', function () {
+  describe('VirtualHostAgent', function () {
+    /* jshint expr:true */
+
+    const ip = '127.0.0.1';
+    const ips = [ip];
+    const port = 2727;
+    const auth = {
+      user: 'admin',
+      pass: 'secret'
+    };
+    const response = {
+      body: {}
+    };
+    const body = {};
+    const parameters = {
+      foo: 'bar'
+    };
+    const credentials = {
+      password: 'secret'
+    };
+    const logs = [{
+      level: 'info',
+      msg: 'foo'
+    }, {
+      level: 'critical',
+      msg: 'bar'
+    }];
+    let version = 2;
+    let protocol = 'http';
+    let pathname = 'foo';
+    let expectedStatus;
+    let api_version = '1.1';
+    let supported_features = ['state', 'lifecycle', 'credentials', 'backup', 'multi_tenancy'];
+    let url;
+    let agent;
+    let requestStub;
+    const instanceId = '06e18533-48c9-4533-8686-603cbfa61c7e';
+
+    function createAgent(opts) {
+      return new VirtualHostAgent(_.assign({
+        version: version,
+        auth: auth,
+        protocol: protocol,
+        port: port
+      }, opts));
+    }
+
+    function createUrl(pathname) {
+      return `${protocol}://${ip}:${port}/v${version}/${pathname}`;
+    }
+
+    beforeEach(function () {
```

Comment:
you dont need to stub beforeEach and restore afterEach. You can have stubs at `before()` & `after()` at test case level.